transport/internet/config.proto

```diff
@@ -121,4 +121,6 @@ message SocketConfig {
   bool tcp_no_delay = 18;
 
   bool tcp_mptcp = 19;
+
+  map<int32, int32> customSockopt = 20;
```

Comment:
I think it is a good idea, please make it more generic that user should be able to define level (2nd param in .Setsockopt()) as well. Also it should support different value types.. Maybe something like
message Custom {
  int64 level;
  int64 opt;
  int32 type;
  string value;
}